### PR TITLE
Add table showing performance tips and polish ui.

### PR DESCRIPTION
**Pull request by jacob314**
Use yellow icons.
Perf Panel tweaks around being clever about tweaking the layout depending on the aspect ratio and size.
Att a table

### CODE REVIEW

File: src/io/flutter/inspector/FrameRenderingDisplay.java [hunk 1]
```diff
@@ -35,7 +35,7 @@ public static JPanel createJPanelView(Disposable parentDisposable, FlutterApp ap
     final JPanel panel = new JPanel(new StackLayout());
     panel.setDoubleBuffered(true);
     panel.setPreferredSize(new Dimension(-1, HeapDisplay.PANEL_HEIGHT));
-    panel.setMaximumSize(new Dimension(Short.MAX_VALUE, HeapDisplay.PANEL_HEIGHT));
```
Comment (jacob314) [inline src/io/flutter/inspector/FrameRenderingDisplay.java]: Allow the FPS display to resize to take better advantage of the available area.

File: src/io/flutter/inspector/WidgetPerfTipsPanel.java [hunk 2]
```diff
@@ -22,23 +24,21 @@
...
-// TODO(jacobr): display a table of all widget perf stats in this panel along
-// with the summary information about the current performance stats.
+import java.util.*;
 
 /**
- * Panel displaying basic information on Widget perf for the currently selected
- * file.
+ * Panel displaying performance tips for the currently visible files.
  */
-public class WidgetPerfPanel extends JPanel {
```
Comment (jacob314) [inline src/io/flutter/inspector/WidgetPerfTipsPanel.java]: Simplified this to just focus on the perf tips as the messages are now displayed inline in the editor.

File: src/io/flutter/perf/EditorPerfDecorations.java [hunk 3]
```diff
@@ -267,13 +265,6 @@ public void clear() {
  * required.
  */
 class PerfGutterIconRenderer extends GutterIconRenderer {
-  static final AnimatedIcon RED_PROGRESS = new RedProgress();
```
Comment (jacob314) [inline src/io/flutter/perf/EditorPerfDecorations.java]: Moved this over to an Icons class as the Icons are now reused here and in the table.

**Comment by jacob314** [comment]
Here is a screenshot of what the table looks like.
small_table

File: src/io/flutter/view/WidgetPerfTable.java [hunk 4]
```diff
@@ -0,0 +1,518 @@
...
+        final Object selectedRow = getTreeModel().getRowValue(selectedRowIndex);
+        if (selectedRow instanceof DefaultMutableTreeNode) {
+          final DefaultMutableTreeNode selectedRowNode = (DefaultMutableTreeNode)selectedRow;
+          final SlidingWindowStatsSummary selectedStats = (SlidingWindowStatsSummary)selectedRowNode.getUserObject();
+          if (selectedStats != null) {
+            lastSelectedLocation = selectedStats.getLocation();
+          }
+        }
+      }
+      // TOOD(jacobr): avoid updating the model when
```
Comment (kenzieschmoll) [inline src/io/flutter/view/WidgetPerfTable.java]: Avoid updating the model when...??

File: src/io/flutter/view/WidgetPerfTable.java [hunk 5]
```diff
@@ -0,0 +1,518 @@
...
+        }
+        final int endChildIndex = childIndex;
+        final ArrayList<TreeNode> nodesRemoved = new ArrayList<>();
+        final TIntArrayList indicesRemoved = new TIntArrayList();
+        // Gather nodes to remove.
+        for (int j = endChildIndex; j < root.getChildCount(); j++) {
+          nodesRemoved.add(root.getChildAt(j));
+          indicesRemoved.add(j);
+        }
+        // Actual remove nodes.
```
Comment (kenzieschmoll) [inline src/io/flutter/view/WidgetPerfTable.java]: Actual -> Actually?

File: src/io/flutter/view/WidgetPerfTable.java (hunk 4 above)
Comment (jacob314) [reply src/io/flutter/view/WidgetPerfTable.java]: removed incomplete comment as I fixed it but forgot to update.

File: src/io/flutter/view/WidgetPerfTable.java (hunk 5 above)
Comment (jacob314) [reply src/io/flutter/view/WidgetPerfTable.java]: yep.

**Pull request merged**